src/reader/mod.rs

```diff
@@ -759,6 +818,60 @@ impl<R> Reader<R> {
     pub const fn decoder(&self) -> Decoder {
         self.state.decoder()
     }
+
+    /// Get the direct access to the underlying reader, but tracks the amount of
+    /// read data and update [`Reader::buffer_position()`] accordingly.
+    ///
+    /// # Example
+    ///
+    /// This example demonstrates, how it is possible to read embedded binary data.
+    /// Such XML documents are exist in the wild.
```

Comment:
This example demonstrates how to read stream raw bytes from an XML document.  This could be used to implement streaming read of text, or to read raw binary bytes embedded in an XML document.  (Documents with embedded raw bytes are not valid XML, but XML-derived file formats exist where such documents are valid).